internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -223,11 +224,18 @@ func (md *machineDeployment) deployCreateMachinesForGroups(ctx context.Context,
 		}
 
 		services := groupConfig.AllServices()
+		autostopSetting := fly.MachineAutostopSuspend
 		for _, s := range services {
-			if s.AutoStopMachines != nil && *s.AutoStopMachines {
-				groupsWithAutostopEnabled[name] = true
+			if s.AutoStopMachines != nil && *s.AutoStopMachines < autostopSetting {
+				autostopSetting = *s.AutoStopMachines
 			}
 		}
+		switch autostopSetting {
```

Comment:
I think there is an edge case when `s.AutoStopMachines == nil` that defaults to `MachineAutostopSuspend` when it should be `MachineAutostopOff`